Source/WebKit/UIProcess/Launcher/glib/BubblewrapLauncher.cpp

```diff
@@ -390,6 +390,27 @@ static const char* environmentVariableValue(const char* name, const char* defaul
     return value ? value : defaultValue;
 }
 
+static void bindUsrMerge(Vector<CString>& args)
+{
+    static const std::array<String, 5> usrMergedDirectories = {
+        "/bin"_s,
+        "/lib"_s,
+        "/lib32"_s,
+        "/lib64"_s,
+        "/sbin"_s
+    };
+    for (auto& directory : usrMergedDirectories) {
+        auto fileType = FileSystem::fileType(directory);
+        if (fileType == FileSystem::FileType::SymbolicLink) {
+            auto symLink = makeString("/usr"_s, directory, '/');
+            args.appendVector(Vector<CString> { "--symlink", directory.utf8(), symLink.utf8() });
+        } else if (fileType == FileSystem::FileType::Directory) {
+            auto symLink = makeString("/usr"_s, directory);
+            args.appendVector(Vector<CString> { "--ro-bind-try", symLink.utf8(), symLink.utf8() });
```

Comment:
This fails on non-usrmerged systems, e.g. if /bin is not a symlink then it won't be bound.

What I would do is bind normally here instead:

```suggestion
            args.appendVector(Vector<CString> { "--ro-bind", directory.utf8(), directory.utf8() });
```

That will take care of binding any paths not under /usr, like /bin or /lib. But we also need to bind the paths under /usr, like /usr/bin or /usr/lib. We could do that above, in the `if (fileType == FileSystem::FileType::SymbolicLink)` condition, but that would fail if the symlinks ever get removed (you never know, it could happen 50 years from now!), so probably best to just restore the use of --ro-bind-try that you removed for the locations under /usr?

Reply:
The reason the current wip patch is like this is that `--ro-bind*` and `--symlink` don't seem to play well together. If I keep both I get this `bwrap: Can't make symlink at /bin: destination exists and is not a symlink`.

I think this needs the attention of someone actually familiar with the `bwrap` options.

Reply:
The options do exactly what you tell them to. It's either create a symlink or mount a directory. The error message indicates exactly what is wrong: this is a usrmerged system and /bin is supposed to be a symlink to /usr/bin, so BubblewrapLauncher.cpp shouldn't be trying to --ro-bind the host /bin. But we do need to --ro-bind it on non-usrmerged systems. We need to do _one of_ create symlink or bind host directory -- never both -- for each of these locations. That error message indicates your code is doing both. It's probably a simple mistake?

Reply:
Also: I notice the existing sandbox intentionally skips binding /bin except in the `enableDebugPermissions` case. We should probably maintain that behavior and just altogether remove /bin and /sbin from this function. We could create the symlinks in the `enableDebugPermissions` codepath if you want, though.

Reply:
You realize you contradict yourself then? You said "so probably best to just restore the use of --ro-bind-try that you removed for the locations under /usr?" and now "We need to do one of create symlink or bind host directory -- never both -- for each of these locations".

Reply:
Use --ro-bind-try in the non-usrmerge condition, when /bin is not a symlink.